core/src/main/java/io/lionweb/lioncore/java/language/LionCoreBuiltins.java

```diff
@@ -1,27 +1,36 @@
 package io.lionweb.lioncore.java.language;
 
-import io.lionweb.lioncore.java.serialization.JsonSerialization;
+import io.lionweb.lioncore.java.LionWebVersion;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nonnull;
 
 public class LionCoreBuiltins extends Language {
-  private static final LionCoreBuiltins INSTANCE = new LionCoreBuiltins();
+  // We may have one instance of this class per LionWeb version, and we initialize
+  // them lazily
+  private static final Map<LionWebVersion, LionCoreBuiltins> INSTANCES = new HashMap<>();
 
   /** This is private to prevent instantiation and enforce the Singleton pattern. */
-  private LionCoreBuiltins() {
-    super("LionCore_builtins");
+  private LionCoreBuiltins(@Nonnull LionWebVersion lionWebVersion) {
+    super(lionWebVersion, "LionCore_builtins");
     setID("LionCore-builtins");
```

Comment:
This means we have more than one node with the same id. Do we want this? It violates one of our fundamental assumptions -- every node has a unique id.

We also have an (unresolved) issue on this: https://github.com/LionWeb-io/specification/issues/279

Reply:
I thought that the specs dictated these IDs, and I think this should be the case. Otherwise when we load models from different sources in the same repository we may have confusion as we refer to nodes everyone is aware of (e.g., the node for `Concept`) through different IDs.

For now however clashing ids should not be a problem because of the decision of considering different versions of lionweb incompatible, so we cannot mix together in the same "universe" nodes from different versions.

Reply:
This will be handled in https://github.com/LionWeb-io/lionweb-java/issues/172